scylla.go

```diff
@@ -490,10 +493,7 @@ func (p *scyllaConnPicker) Put(conn *Conn) {
 			until := time.Now().Add(scyllaShardAwarePortFallbackDuration)
 			p.disableShardAwarePortUntil.Store(until)
 
-			// Connections to shard-aware port do not influence how shards
-			// are chosen for the non-shard-aware port, therefore it can be
-			// closed immediately
-			closeConns(conn)
+			return errors.New("scylla: falling back to non-shard-aware port due to wrong shard assignment")
```

Comment:
Here you need to tell why connection was not added to the pool
```suggestion
			return fmt.Errorf("connection landed on %d shard that already has connection", shard)
```